ide/coqide/coqOps.ml

```diff
@@ -567,6 +589,15 @@ object(self)
           | _ -> None in
         try Some (Doc.find_map document finder)
         with Not_found -> None in
+      (* When a command generates a warning or error, the warning/error
+         message is returned before the Processed message that provides the
+         sentence id.  There can only be a single sentence in the document
+         with an id of None, which must be at the tip.  A protocol design
+         weakness.  Not an issue for tactics. *)
+      let sentence = match sentence with
+        | None -> (try Some (None, Doc.tip_data document) with Doc.Empty -> None)
+        | Some (sid, sent) -> sentence
```

Comment:
This seems nice! Should go in its own PR IMHO tho.

Reply:
As I say in the PR description, "[the second commit is] optional for 8.17, I can split it out if desired."